src/interpolation_transfer.jl

```diff
@@ -194,3 +194,49 @@ function transfer_blochwave(ψ_in, basis_in::PlaneWaveBasis{T},
         transfer_blochwave_kpt(ψ_in[ik], basis_in, basis_in.kpoints[ik], basis_out, kpt_out)
     end
 end
+
+"""
+Return Integer with defined tolerance.
+"""
+function _isinteger(x; tol=nothing)
+    isnothing(tol) && return isinteger(x)
+    abs(round(x) - x) < tol
+end
+
+"""
+Shift the representation of a Bloch wave from function from ``k``-point `kpt_in` to
+``k``-point `kpt_out`, where `kpt_in - kpt_out` is a vector in the reciprocal lattice.
+"""
+function shift_blochwave_kpt(basis, ψk_in, kpt_in, q)
+    tol = 100*eps(eltype(q))
+    # coordinates = [kpt.coordinate for kpt in basis.kpoints]
+    # coordinates = map(kpt.coordinate, basis.kpoints)
+    coordinates = getfield.(basis.kpoints, :coordinate)
+    kpt_shifted = kpt_in.coordinate + q
+
+    #index = findall(e -> e == kpt_shifted, coordinates)
+    index = findall(e -> ≈(e, kpt_shifted, atol=tol), coordinates)
+    @assert length(index) ≤ 1
+
+    if !isempty(index)
+        kpt_out = basis.kpoints[index[1]]
+        return transfer_blochwave_kpt(ψk_in, basis, kpt_in, basis, kpt_out)
+    end
+
+    differences = [c - kpt_shifted for c in coordinates]
+    index = findall(e -> prod(_isinteger.(abs.(e); tol=tol)), differences)
+    @assert length(index) ≤ 1
+
+    if !isempty(index)
+        kpt_out = basis.kpoints[index[1]]
+        ψk_out = zeros(eltype(ψk_in), length(kpt_out.G_vectors), size(ψk_in, 2))
+        shift = cis2pi.(-dot.(Ref(differences[index[1]]), r_vectors(basis)))
+        for n in 1:size(ψk_in, 2)
+            ψk_out[:, n] = r_to_G(basis, kpt_out, shift .* G_to_r(basis, kpt_in, ψk_in[:, n]))
+        end
```

Comment:
this is fine for debug, but do that without going to real space, just by manipulation of the G vectors. it should already be somewhere in the file, we can discuss if not